### DISCUSSION TITLE
Mayo as docker image

### DISCUSSION THREAD

**Discussion by nikhil-convrseai** [fougue/mayo#253] [Ideas]
I want to run mayo in a docker container with only it's cmd capabilities, how can we achieve that?

**Comment by HuguesDelorme** [comment]
What is the Linux distro you will be using within Docker?
If it's Debian-based, eg Ubuntu then follow the [build instructions for Debian](https://github.com/fougue/mayo/wiki/Build-instructions-for-Debian)
You can also directly use Mayo AppImage instead(see the [Releases](https://github.com/fougue/mayo/releases) page)

Use Mayo application as expected
You may need to run `Xvfb $DISPLAY -screen 0 1280x1024x24 &` from the command line if there's no X session running

**Reply by nikhil-convrseai** [reply]
I tried build for Debian inside docker, but it takes forever and doesn't complete whereas on server it is building very quickly.

here is my docker file

```# Start from a Debian base image
FROM debian:bullseye


ENV MAYO_GIT_URL=https://github.com/fougue/mayo.git

# Install dependencies
RUN apt-get update && apt-get install -y \
    build-essential \
    git \
    qtbase5-dev \
    qttools5-dev \
    libocct-data-exchange-dev \
    libocct-foundation-dev \
    libocct-modeling-algorithms-dev \
    libocct-modeling-data-dev \
    libocct-ocaf-dev \
    libocct-visualization-dev \
    && rm -rf /var/lib/apt/lists/*

# Clone the Mayo repository
RUN mkdir -p /home/ubuntu

WORKDIR /home/ubuntu
RUN git clone $MAYO_GIT_URL 
RUN mkdir mayo-build 
WORKDIR /home/ubuntu/mayo-build
# Set environment variables using RUN command
RUN CASCADE_INC_DIR=$(dpkg -L libocct-foundation-dev | grep -i "Standard_Version.hxx" | sed "s/\/Standard_Version.hxx//i") && \
    CASCADE_LIB_DIR=$(dpkg -L libocct-foundation-dev | grep -i "libTKernel.so" | sed "s/\/libTKernel.so//i") && \
    ASSIMP_INC_DIR=$(dpkg -L libassimp-dev | grep -i "version.h" | sed "s/\/version.h//i") && \
    ASSIMP_LIB_DIR=$(dpkg -L libassimp-dev | grep -i "libassimp.so" | sed "s/\/libassimp.so//i") && \
    qmake ../mayo \
      CASCADE_INC_DIR=$CASCADE_INC_DIR \
      CASCADE_LIB_DIR=$CASCADE_LIB_DIR \
      ASSIMP_INC_DIR=$ASSIMP_INC_DIR   \
      ASSIMP_LIB_DIR=$ASSIMP_LIB_DIR

RUN make -j 10 -k || make  VERBOSE=1
```

**Reply by HuguesDelorme** [reply]
You said "it does not complete" but on which step or file is it blocking?
Compilation on Docker image takes inherently more time

**Reply by stellarpower** [reply]
@HuguesDelorme I've built successfully inside arch just following the build instructions. However, I was wondering if it would be worth committing and pushing up the image used for CI - either with dependencies or with the full application - just a thought, for development speed.

**Reply by HuguesDelorme** [reply]
@stellarpower
It could be a good idea to upload some artifacts at each CI build, but this is currently not that much possible because of GitHub "shared storage usage"
Storage quota is 512MB and beyond that limit GitHub wants the account owner to pay for extra MB
That limit would be reached early each month, maybe I should ask for donations to support this

**Reply by stellarpower** [reply]
Fair enough, probably not worth that! Presumably docker hub would be an option(?)
